### PR TITLE
[Go-gen] Struct handlers

### PR DESCRIPTION
**Pull request by c-harding**
Here come the structs! Adds functions for checking parentage, the handler functions, and the kong registration functions.

Once #392 is merged, it’s +1,172 -56, which is +219 -56 of code plus 953 new lines of Go code (all very repetitive and basically the same as what would be generated if the structs were duplicated).

Currently failing the Deliveroo integration test due to an existing issue, as shown in #386.

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
Alright cool cool

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainDeleteHandlerGenerator.scala [hunk 1]
```diff
@@ -19,7 +20,11 @@ object GoServiceMainDeleteHandlerGenerator {
       "input",
     )
 
-  private def generateDAOCallBlock(block: AttributesRoot, metricSuffix: Option[String]): String =
+  private def generateDAOCallBlock(
+    block: AttributesRoot,
+    parent: Option[ServiceRoot],
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainDeleteHandlerGenerator.scala]: Is this param used?

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainReadHandlerGenerator.scala [hunk 2]
```diff
@@ -3,17 +3,23 @@ package temple.generate.server.go.service.main
...
 
 import scala.Option.when
 import scala.collection.immutable.ListMap
 
 object GoServiceMainReadHandlerGenerator {
 
-  private def generateDAOCallBlock(block: AttributesRoot, metricSuffix: Option[String]): String =
+  private def generateDAOCallBlock(
+    block: AttributesRoot,
+    parent: Option[ServiceRoot],
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainReadHandlerGenerator.scala]: Ditto here?

**Review by jaylees14 — APPROVED** [review]
Cool, much easier to review when it's just the additions. Generated code looks good, would normally do some manual testing but I'll save that for `temple test`

**Pull request merged**